src/dodal/devices/motors.py

```diff
@@ -2,60 +2,173 @@
 
     Parameters
     ----------
     prefix:
-        EPICS PV (Common part up to and including :).
+        Common part of the EPICS PV for all motors, including ":".
     name:
-        name for the stage.
-    infix:
-        EPICS PV, default is the ("X", "Y", "Z").
-    Notes
-    -----
-    Example usage::
-        async with init_devices():
-            xyz_stage = XYZPositioner("BLXX-MO-STAGE-XX:")
-    Or::
-        with init_devices():
-            xyz_stage = XYZPositioner("BLXX-MO-STAGE-XX:", infix = ("A", "B", "C"))
+        Name of the stage, each child motor will be named "{name}-{field_name}"
+    *_infix:
+        Infix between the common prefix and the EPICS motor record fields for the field.
+
+    """
+
+    def __init__(
+        self, prefix: str, name: str = "", x_infix: str = "X", theta_infix: str = "A"
+    ):
+        with self.add_children_as_readables():
+            self.x = Motor(prefix + x_infix)
+            self.theta = Motor(prefix + theta_infix)
+        super().__init__(name=name)
+
+
+class XYStage(StandardReadable):
+    """
+
+    ophyd_async x y motor stage, combining 2 Motors, mutually perpendicular to the beam.
+
+    Parameters
+    ----------
+    prefix:
+        Common part of the EPICS PV for all motors, including ":".
+    name:
+        Name of the stage, each child motor will be named "{name}-{field_name}"
+    *_infix:
+        Infix between the common prefix and the EPICS motor record fields for the field.
+
+    """
+
+    def __init__(
+        self, prefix: str, name: str = "", x_infix: str = "X", y_infix: str = "Y"
+    ):
+        with self.add_children_as_readables():
+            self.x = Motor(prefix + x_infix)
+            self.y = Motor(prefix + y_infix)
+        super().__init__(name=name)
+
+
+class XYZStage(XYStage):
+    """
+
+    ophyd_async x y z motor stage, combining 3  mutually perpendicular Motors.
+
+    Parameters
+    ----------
+    prefix:
+        Common part of the EPICS PV for all motors, including ":".
+    name:
+        Name of the stage, each child motor will be named "{name}-{field_name}"
+    *_infix:
+        Infix between the common prefix and the EPICS motor record fields for the field.
 
     """
 
     def __init__(
         self,
         prefix: str,
         name: str = "",
-        infix: tuple[str, str, str] = ("X", "Y", "Z"),
+        x_infix: str = "X",
+        y_infix: str = "Y",
+        z_infix: str = "Z",
     ):
         with self.add_children_as_readables():
-            self.x = Motor(prefix + infix[0])
-            self.y = Motor(prefix + infix[1])
-            self.z = Motor(prefix + infix[2])
-        super().__init__(name=name)
+            self.z = Motor(prefix + z_infix)
+        super().__init__(prefix, name, x_infix, y_infix)
 
 
-class SixAxisGonio(XYZPositioner):
+class XYZThetaStage(XYZStage):
+    """
+
+    ophyd_async x y z theta motor stage, combining 3  mutually perpendicular Motors and
```

Comment:
Nit: Remove additional space, same in all other docstrings
```suggestion
    ophyd_async x y z theta motor stage, combining 3 mutually perpendicular Motors and
```